node/zombienet-backchannel/src/lib.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright 2017-2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+/// Provides a backchannel to zombienet test runner.
+use std::env;
+use futures_util::{StreamExt, stream::SplitStream};
+use tokio::net::TcpStream;
+use tokio_tungstenite::{MaybeTlsStream, WebSocketStream, connect_async};
+use parity_scale_codec as scale_codec;
+
+#[macro_use]
+extern crate lazy_static;
+
+mod errors;
+use errors::BackchannelError;
+
+#[derive(Debug)]
+pub struct ZombienetBackchannel {
+    pub inner: SplitStream<WebSocketStream<MaybeTlsStream<TcpStream>>>
+}
+
+lazy_static! {
+    static ref BACKCHANNEL_HOST: String = env::var("BACKCHANNEL_HOST").unwrap_or_else(|_| "backchannel".to_string());
+    static ref BACKCHANNEL_PORT: String = env::var("BACKCHANNEL_PORT").unwrap_or_else(|_| "3000".to_string());
+}
+
+
+impl ZombienetBackchannel {
+    pub async fn connect() -> Result<ZombienetBackchannel,BackchannelError> {
+        let ws_url = format!("ws://{}:{}/ws", *BACKCHANNEL_HOST, *BACKCHANNEL_PORT);
+        log::debug!("Connecting to : {}",&ws_url);
+        let (ws_stream, _) = connect_async(ws_url).await.map_err(|_| BackchannelError::CantConnectToWS)?;
+        let (_write, read) = ws_stream.split();
+
+        Ok( ZombienetBackchannel {
```

Comment:
Idea: could we use a slighly different pattern where we use an `fn init` to setup the initial connection - assuming this is at the very beginning of the `malus` invocation. That then stores the `sink` and the `source` part of `ws` so we can use both? The `sink` is easily copyable, the `source` would need some additional thoughts. This would allow to consume messages in multiple places without many extra changes across the code base.

Even if this a bit further out, we should try to make the API compatible in that direction, i.e. move some logic from `lazy_static` to a global `fn initialize` . Happy to discuss this further.

Reply:
Hi @drahnr, thanks for the feedback! I like the idea of use a `init` fn to initialize the `ws` connection and store both parts to use later. If we want to follow this path I think we can use a `broadcaster` to connect the `source` part of the `ws` and forward the message to multiple consumers.

Also, a more general question about how the `malus` will consume this information, Does the `malus` will need to `store` the received values to reads when is needed right? I was thinking in a `hashmap` as we store in the `zombienet` part and follow that direction another option is to pass to the `init` fn the hashmap behind an `RwLock` since most of the time the receiving part of the `ws` will `write` new values and the `malus` actor will `read` (since now all the messages received in the zombienet part are forwarded to all connected actors). 
**But** it's just and idea and I'm not sure if this is the direction we want follow.  Happy to discuss this further :)

Thanks!